ibis/backends/base/sql/compiler/query_builder.py

```diff
@@ -536,6 +539,9 @@ def to_ast(cls, node, context=None):
         if isinstance(node, ir.Expr):
             node = node.op()
 
+        if cls.rewrites:
+            node = node.replace(cls.rewrites)
```

Comment:
I looked into adding these generically in the `Backend` instead of in `Compiler`. However, here felt better since it was a much smaller code change. `to_ast` is called in many different places, trying to unify all those locations to also hit a backend's rewrite rules would be a larger refactor. IMO we can move rewrites into the `Backend` class later if needed.

Reply:
I suspect this will be less of a refactor when more backends are sqlglot-based.

Reply:
Agree that the compiler is the best place for these at the moment.